File: 1100/1631B/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin>>t;

    while(t--) 
    {
        int n;
        cin>>n;

        vector<int> a(n);
        for (int i = 0; i < n; i++)
            cin>>a[i];

        int ans = 0;

        // Use a for loop just like your original Java code to handle i-- automatically
        for(int i=n-2; i>=0; i--) 
        {
            // Skip elements matching the target value at the end
            while(i >= 0 && a[i] == a[n - 1])
                i--;

            if(i >= 0)
                ans++;

            // Calculate the length of the matched segment (b)
            int b = (n - 1) - i;
            
            // Jump `i` back
            i = i - b + 1;
        }

        cout << ans << "\n";
    }

    return 0;
}

File: 1100/1631B/main.java
import java.util.*;
public class main
{
	public static void main(String args[]) 
    {
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
 
		while(t-- > 0) 
        {
			int n = sc.nextInt();
			int a[] = new int[n];
			for(int i=0;i<n;i++)
				a[i] = sc.nextInt();
 
			int ans = 0;
			int c = 0;
			for(int i=n-2;i>=0;i--) 
            {
				while(i >= 0 && a[i] == a[n-1])
					i--;
 
				if(i >= 0)
					ans++;
 
				int b = n-1-(i);
				i = i-b+1;
			}
 
			System.out.println(ans);
		}
	}
}